### ISSUE TITLE
Changing/Improving resolution, delay,buffer, fps  of live preview stream (capture-movie)

### ISSUE THREAD

**Issue by QuditWolf** [gphoto/gphoto2#405]
Camera: Nikon D3400
OS: ArchLinux
I am trying to stream from camera and use it as a webcam. ( Without buying an external video card).
gphoto is all set up and works
The images taken are at highest res.
But when the output of movie capture is piped to ffmpeg it detects the input res as something lower and feed is stuttery.
How can this be solved using gphoto2 or ffmpeg?

Edit: I did try to improve resolution by set config but that doesn't seem to improve the video resolution and fps..

cli command:
gphoto2 --stdout --capture-movie | ffmpeg -i - -vcodec rawvideo -pix_fmt yuv420p -threads 0 -f v4l2 /dev/video2

I know this is possible as Entangle which is a GUI using gphoto does this perfectly along with live preview and live tweaking of parameters such as resolution, and camera settings.... (https://gitlab.com/entangle/entangle)

PS: Amazing piece of code though ! A good webcam or an external vieo capture card costs a lot . When we can easily use existing good cameras..🙂

**Comment by vandenheuvel** [comment]
@QuditWolf were you able to get your D3400 to work?

**Comment by msmeissn** [comment]
For one thing "movie" capture only streams the liveview , which is smaller than FULL size images.

**Comment by QuditWolf** [comment]
Sorry for not checking this earlier.
@vandenheuvel It works on D3400 in the minimal sense but I could not get the higher resolutions to work.
And @msmeissn I got that. but the resolution of the live stream I get is lower(somewhat near 600x400). Whilst If I preview with entangle which I believe is a wrapper on libgphoto2, the live view works just fine.

**Comment by QuditWolf** [comment]
@msmeissn,
yeah I think the movie capture speed on my camera (Nikon D3400) can output 720p... Also I can see a quite good quality stream on apps like entangle (which uses libgphoto2).

piping it to ffmpeg thru stdout, resolution is stuck at 640x424, while camera can ouput ~720p as preview...

Is there someway to specify a rate at which to capture frames? or is the problem lying in ffmpeg reading from stdout and misdetecting the format/frame rate/resolution(have no idea, shooting arrows in the dark)?

here is how I am using it right now.  or any better way to do this?
do there exist apps to do this?(I couldn't find any)
```
$  gphoto2 --stdout --capture-movie | ffmpeg -i - -vcodec rawvideo -pix_fmt yuv420p -threads 0 -f v4l2 /dev/video2 -hide_banner
Capturing preview frames as movie to 'stdout'. Press Ctrl-C to abort.
[mjpeg @ 0x55b0ec457400] Format mjpeg detected only with low score of 25, misdetection possible!
Input #0, mjpeg, from 'pipe:':
  Duration: N/A, bitrate: N/A
  Stream #0:0: Video: mjpeg (Baseline), yuvj422p(pc, bt470bg/unknown/unknown), 640x424, 25 tbr, 1200k tbn
Stream mapping:
  Stream #0:0 -> #0:0 (mjpeg (native) -> rawvideo (native))
[swscaler @ 0x55b0ec481dc0] [swscaler @ 0x55b0ec4906c0] deprecated pixel format used, make sure you did set range correctly (this line appears atleast 30-40 times omitted here...)
Output #0, video4linux2,v4l2, to '/dev/video2':
  Metadata:
    encoder         : Lavf59.27.100
  Stream #0:0: Video: rawvideo (I420 / 0x30323449), yuv420p(tv, bt470bg/unknown/unknown, progressive), 640x424, q=2-31, 81408 kb/s, 25 fps, 25 tbn
    Metadata:
      encoder         : Lavc59.37.100 rawvideo
^CCtrl-C pressed ... Exiting.ze=N/A time=00:00:07.40 bitrate=N/A speed=7.26x    
Movie capture finished (252 frames)
frame=  251 fps=173 q=-0.0 Lsize=N/A time=00:00:10.04 bitrate=N/A speed=6.93x    
video:99772kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Exiting normally, received signal 2.

```

Would using libgphoto2 in c or python to write to the v4l2device be a better option? What would be the easiest way of doing that?

Thanks for your wonderful project,
Quditwolf
